### DISCUSSION TITLE
How do I get the current IPython / Jupyter Notebook name?

### DISCUSSION THREAD

**Discussion by belonesox** [sagemathinc/cocalc#6614] [Q&A] [answered]
How do I get the current IPython / Jupyter Notebook name?

There is a lot of ways to get it under Jupyter Notebook/Lab/VS Code/…  see  [How do I get the current IPython / Jupyter Notebook name](https://stackoverflow.com/questions/12544056/how-do-i-get-the-current-ipython-jupyter-notebook-name). 

But none of them works with cocalc. :(

Probably better will be adding  support for that to [ipynbname ](https://github.com/msm1089/ipynbname) or [ipyparams](https://github.com/nbgallery/ipyparams), but first we should fix some public JS API for that.

**Comment by williamstein** [comment]
I think what you are asking for is currently not possible in cocalc.

That said, I was working on very relevant nearby code this week, so I bet I can implement it in a few minutes.

See https://github.com/sagemathinc/cocalc/issues/6615

**Reply by williamstein** [reply]
Update -- use the COCALC_JUPYTER_KERNELNAME environment variable.  It's been in cocalc for a long time.

**Reply by belonesox** [reply]
![image](https://user-images.githubusercontent.com/1609739/230227769-25373ce6-4b03-49da-99ed-78e00c11a03e.png)

I mean Jupyter notebook name/path…  Or I shoud use it to patch [ipynbname](https://github.com/msm1089/ipynbname)?

**Reply by williamstein** [reply, marked as answer]
You should definitely patch ipynbname to support cocalc if you want.

In any case, just look at all the env vars that start with COCALC and you'll answer your own question above.

**Reply by belonesox** [reply]
Okay, let's use `os.environ['COCALC_JUPYTER_FILENAME']` as the interface.
